docs/metrics/bertscore.md

```diff
@@ -0,0 +1,117 @@
+
+??? question "Recall: BERT & Textual Embeddings"
+    BERT (**B**idirectional **E**ncoder **R**epresentations from **T**ransformers) is a popular language model used to generate embeddings from words and phrases. Textual embeddings are learned dense token representations that capture the semantic and contextual information of words and phrases in a continuous vector space. In a perfect embedding space, similar words are grouped together while words that share little relationship are distanced. For a deeper dive into BERT and textual embeddings, feel free to refer to the [original paper](https://arxiv.org/pdf/1810.04805.pdf).
+    <center>![Embedding Space Image](../assets/images/bert_vector_space.png){: style="height:20em;width:auto"}</center>
+
+## Implementation Details
+BERTScore is a collection of three metrics - BERT-Precision, BERT-Recall, and BERT-F1. As the names imply, BERT-Precision measures how well the candidate texts avoid introducing irrelevant content. BERT-Recall measures how well the candidate texts avoid omitting relevant content. BERT-F1 is a combination of both Precision and Recall to measure how well the candidate texts capture and retain relevant information from the reference texts.
+
+### Calculating BERTScore
+Given a reference sentence, $x = \langle x_1, x_2, ..., x_n \rangle$, and machine-generated sentence, $\hat{x} = \langle\hat{x}_1, \hat{x}_2, ..., \hat{x}_m\rangle$, we first use BERT to generate a sequence of word embeddings for both reference and candidate sentences.
+
+$$
+\begin{align}
+    & BERT(\langle x_1, x_2, ..., x_n \rangle) = \langle \mathbf{x_1}, \mathbf{x_2}, ..., \mathbf{x_n} \rangle \\
+    & BERT(\langle \hat{x}_1, \hat{x}_2, ..., \hat{x}_m \rangle) = \langle \mathbf{\hat{x}_1}, \mathbf{\hat{x}_2}, ..., \mathbf{\hat{x}_m} \rangle
+\end{align}
+$$
+
+<center><p style="font-size:small;">Note that we will use <b>bold</b> text to indicate vectors, like a word embedding</p></center>
+
+To measure the similarity between two individual embeddings, we will use the cosine similarity which is defined as:
+
+$$
+\text{similarity}(\mathbf{x_i}, \mathbf{\hat{x}_j}) = \frac{\mathbf{x_i}^\top \mathbf{\hat{x}_j}}{||\mathbf{x_i}||\space||\mathbf{\hat{x}_j}||}
+$$
+
+which simply reduces to $\mathbf{x_i}^\top \mathbf{\hat{x}_j}$ since both $\mathbf{x_i}$ and $\mathbf{\hat{x}_j}$ are pre-normalized. With these definitions, we can now calculate the BERT precision, recall, and f1-score.
+
+#### BERT Precision
+
+$$
+P_\text{BERT} = \frac{1}{|\hat{x}|} \sum_{\mathbf{\hat{x}_j} \in \hat{x}} \underbrace{\max_{\mathbf{x_i} \in x}\overbrace{\mathbf{x_i}^\top \mathbf{\hat{x}_j}}^\text{cosine similarity}}_\text{greedy matching}
```

Comment:
would prefer `BERT_prec` or something similar instead of `P_BERT`, and for the other formulas